pkg/component/controlplane/control_plane.go

```diff
@@ -0,0 +1,64 @@
+package controlplane
+
+import (
+	"fmt"
+	"strings"
+
+	"istio.io/operator/pkg/apis/istio/v1alpha2"
+	"istio.io/operator/pkg/component/feature"
+	"istio.io/operator/pkg/translate"
+	"istio.io/operator/pkg/util"
+)
+
+// IstioControlPlane is an installation of an Istio control plane.
+type IstioControlPlane struct {
+	features []feature.IstioFeature
+	started  bool
+}
+
+// NewIstioControlPlane creates a new IstioControlPlane and returns a pointer to it.
+func NewIstioControlPlane(installSpec *installerv1alpha1.IstioControlPlaneSpec, translator *translate.Translator) *IstioControlPlane {
+	opts := &feature.FeatureOptions{
+		InstallSpec: installSpec,
+		Traslator:   translator,
+	}
+	return &IstioControlPlane{
+		features: []feature.IstioFeature{
+			feature.NewTrafficManagementFeature(opts),
+			feature.NewSecurityFeature(opts),
+			feature.NewPolicyFeature(opts),
+			feature.NewTelemetryFeature(opts),
+			feature.NewConfigManagementFeature(opts),
+			feature.NewAutoInjectionFeature(opts),
+		},
+	}
+}
+
+// Run starts the Istio control plane.
+func (i *IstioControlPlane) Run() error {
+	for _, f := range i.features {
+		if err := f.Run(); err != nil {
+			return err
+		}
+	}
+	i.started = true
+	return nil
+}
+
+// RenderManifest returns a manifest rendered against
+func (i *IstioControlPlane) RenderManifest() (manifest string, errsOut util.Errors) {
+	if !i.started {
+		return "", util.NewErrs(fmt.Errorf("IstioControlPlane must be Run before calling RenderManifest"))
+	}
+	var sb strings.Builder
+	for _, f := range i.features {
+		s, errs := f.RenderManifest()
+		errsOut = util.AppendErrs(errsOut, errs)
```

Comment:
How does the caller know which errors corresponding to which features since they are appended in a list?

Reply:
We have to ensure that the errors have enough context to be meaningful to the caller. That's something that needs to be improved before this can be released. Right now the errors are too cryptic to be useful to end users.

Reply:
Created https://github.com/istio/istio/issues/15225